x/concentrated-liquidity/pool_hooks.go

```diff
@@ -0,0 +1,95 @@
+package concentrated_liquidity
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	types "github.com/osmosis-labs/osmosis/v20/x/concentrated-liquidity/types"
+)
+
+// nolint: unused
+// callPoolActionListener processes and dispatches the passed in message to the contract corresponding to the hook
+// defined by the given pool ID and action prefix (e.g. pool Id: 1, action prefix: "beforeSwap").
+//
+// This function returns an error if the contract address in state is invalid (should be impossible) or if the contract execution fails.
+//
+// Since it is possible for this function to be triggered in begin block code, we need to directly meter its execution and set a limit.
+// If no contract is linked to the hook, this function is a no-op.
+func (k Keeper) callPoolActionListener(ctx sdk.Context, msgBz []byte, poolId uint64, actionPrefix string) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = types.ContractHookOutOfGasError{GasLimit: types.ContractHookGasLimit}
+		}
+	}()
+
+	cosmwasmAddress := k.GetPoolHookContract(ctx, poolId, actionPrefix)
+	if cosmwasmAddress != "" {
```

Comment:
This PR looks great! I only have small nit to do with style. 

```suggestion
	if cosmwasmAddress == "" {
	    return nil
	}
	...
```

I find it's usually best to return early and continue execution than to indent with statement, but this isn't a big issue so feel free to ignore :smile:

Reply:
Great suggestion, thanks! 46aa7f26684c44b455130981a18944005b331e6c